File: java/Plus_Minus.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the plusMinus function below.
    static void plusMinus(int[] arr) {
        int plus_count = 0;
        int zero_count = 0;
        int minus_count = 0;

        for(int num: arr)
        {
            if(num > 0)
                plus_count++;
            else if(num == 0)
                zero_count++;
            else
                minus_count++;
        }

        System.out.printf("%.7f\n", ((float)plus_count/arr.length));
        System.out.printf("%.7f\n", ((float)minus_count/arr.length));
        System.out.printf("%.7f\n", ((float)zero_count/arr.length));

    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        int[] arr = new int[n];

        String[] arrItems = scanner.nextLine().split(" ");
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int i = 0; i < n; i++) {
            int arrItem = Integer.parseInt(arrItems[i]);
            arr[i] = arrItem;
        }

        plusMinus(arr);

        scanner.close();
    }
}

File: scala/PlusMinus.scala
import java.io._
import java.math._
import java.security._
import java.text._
import java.util._
import java.util.concurrent._
import java.util.function._
import java.util.regex._
import java.util.stream._

object Solution {

    // Complete the plusMinus function below.
    def plusMinus(arr: Array[Int]) {
        var plus_count = 0.0d
        var zero_count = 0.0d
        var minus_count = 0.0d
        val arr_size = arr.size

        plus_count = arr.filter(_ > 0).length
        zero_count = arr.filter(_ > 0).length
        minus_count = arr.filter(_ > 0).length

        // for (i <- arr) {
        //     if (i > 0)
        //         plus_count += 1
        //     else if (i == 0)
        //         zero_count += 1
        //     else
        //         minus_count += 1
        // }

        printf("%f\n", (plus_count/arr_size))
        printf("%f\n", (minus_count/arr_size))
        printf("%f\n", (zero_count/arr_size))

    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val n = stdin.readLine.trim.toInt

        val arr = stdin.readLine.split(" ").map(_.trim.toInt)
        plusMinus(arr)
    }
}
